### DISCUSSION TITLE
A 'ADStateful' is not a registered object.

### DISCUSSION THREAD

**Discussion by zxj19910214** [idaholab/moose#27971] [Q&A Getting Started] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (avoid screenshots if possible, triple back quotes before/after pasted text, etc)

### Issue or question about MOOSE

Dear mooseusers,

I try to run an example, it shows the objects not registered. 

The example is:
[Mesh]
  type = GeneratedMesh
  dim = 1
[]

[Variables]
  [./dummy]
  [../]
[]

[Kernels]
  [./diff]
    type = ADMatDiffusion
    variable = dummy
    diffusivity = dummy_prop
  [../]
[]

[Materials]
  [./matprop]
    type = ADStateful
    property_name = dummy_prop
  [../]
[]

[Executioner]
  type = Transient
  num_steps = 1
[]

[Outputs]
  exodus = true
[]

the error is : A 'ADStateful' is not a registered object.

I am writing to ask is there some modules not included when I compile moose?

my makefile is: 

###############################################################################
###############################################################################
#
# Optional Environment variables
# MOOSE_DIR        - Root directory of the MOOSE project
#
###############################################################################
# Use the MOOSE submodule if it exists and MOOSE_DIR is not set
MOOSE_SUBMODULE    := $(CURDIR)/moose
ifneq ($(wildcard $(MOOSE_SUBMODULE)/framework/Makefile),)
  MOOSE_DIR        ?= $(MOOSE_SUBMODULE)
else
  MOOSE_DIR        ?= $(shell dirname `pwd`)/moose
endif

# framework
FRAMEWORK_DIR      := $(MOOSE_DIR)/framework
include $(FRAMEWORK_DIR)/build.mk
include $(FRAMEWORK_DIR)/moose.mk

################################## MODULES ####################################
# To use certain physics included with MOOSE, set variables below to
# yes as needed.  Or set ALL_MODULES to yes to turn on everything (overrides
# other set variables).

ALL_MODULES                 := no

CHEMICAL_REACTIONS          := no
CONTACT                     := no
ELECTROMAGNETICS            := no
EXTERNAL_PETSC_SOLVER       := no
FLUID_PROPERTIES            := no
FSI                         := no
FUNCTIONAL_EXPANSION_TOOLS  := no
GEOCHEMISTRY                := no
HEAT_TRANSFER               := no
LEVEL_SET                   := no
MISC                        := no
NAVIER_STOKES               := no
OPTIMIZATION                := no
PERIDYNAMICS                := no
PHASE_FIELD                 := yes
POROUS_FLOW                 := no
RAY_TRACING                 := no
REACTOR                     := no
RDG                         := no
RICHARDS                    := no
SOLID_MECHANICS             := yes
STOCHASTIC_TOOLS            := no
THERMAL_HYDRAULICS          := no
XFEM                        := yes

include $(MOOSE_DIR)/modules/modules.mk
###############################################################################

# dep apps
APPLICATION_DIR    := $(CURDIR)
APPLICATION_NAME   := puppy
BUILD_EXEC         := yes
GEN_REVISION       := no
include            $(FRAMEWORK_DIR)/app.mk

###############################################################################
# Additional special case targets should be added here

Thank you for your help.

Best regards,
Xiao

### Diagnostics for MOOSE installation issues

[Optional] Output of the diagnostics scripts. To run the script: cd ~/projects/moose/scripts; ./diagnostics.sh

**Comment by GiudGiud** [comment, marked as answer]
Hello

ADStateful is a moose test object. It is not supposed to be used in calculations outside of tests.
It is not part of any module. It can be included but it's a bit of work and not just in the Makefile.
Maybe just move it to your app if you need it (and change the register object call in the source file)

Guillaume

**Reply by zxj19910214** [reply]
Dear Guillaume,

Thank you for answering. I still have one question; recently, I tried to run a test from others, and it has some errors for missing files.
fatal error: ADCZMComputeLocalTractionTotalBase.h: No such file or directory
fatal error: ADSingleVariableReturnMappingSolution.h: No such file or directory

is this the file not compiled from the beginning of installing the software? I could find these files in the moose class index

Thank you for your help in this.
Best regards,
Xiao
